Classes/DDLoggerNames.h

```diff
@@ -0,0 +1,30 @@
+// Software License Agreement (BSD License)
+//
+// Copyright (c) 2010-2018, Deusty, LLC
+// All rights reserved.
+//
+// Redistribution and use of this software in source and binary forms,
+// with or without modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice,
+//   this list of conditions and the following disclaimer.
+//
+// * Neither the name of Deusty nor the names of its contributors may be used
+//   to endorse or promote products derived from this software without specific
+//   prior written permission of Deusty, LLC.
+
+#import <Foundation/Foundation.h>
+
+typedef NSString *const DDLoggerName NS_STRING_ENUM;
+FOUNDATION_EXPORT DDLoggerName DDLoggerNameASL; // DDASLLogger
+FOUNDATION_EXPORT DDLoggerName DDLoggerNameTTY; // DDTTYLogger
+FOUNDATION_EXPORT DDLoggerName DDLoggerNameOSDefault; // DDOSLogger
+FOUNDATION_EXPORT DDLoggerName DDLoggerNameFile; // DDOSLogger
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface DDLoggerNames : NSObject
```

Comment:
There's no need for a class, right? The constants above should be enough.

Reply:
@ffried 
Yes, It could contain some checks in future.

Reply:
Huh? Why would anyone use `DDLoggerNames`?  I don't see the point.

Also, if we don't need it today, we should not include it. If it should ever be needed, it can always be added again.

Reply:
@ffried 

Simple check: 
```objective-c
- (BOOL)isFileLogger:(DDLoggerName);
```

Reply:
I'd write such checks as C-functions. Classes introduce way to much overhead for this:
```
extern inline BOOL DDLoggerNameIsFileLogger(DDLoggerName name);
```

Reply:
Also from the perspective, that this is better portable to Swift, where it could become a property on the `DDLoggerName` struct.

Reply:
@ffried 
Well, structure that have `NSString` as property?

Reply:
Nope, it's a `Bool` in Swift. Also, `NSString` is bridged to `String.

Reply:
This is what it should be mapped to as Swift interface:
```
public struct DDLoggerName: Hashable, Equatable, RawRepresentable {
    public init(_ rawValue: String)
    public init(rawValue: String)
}

public extension DDLoggerName {
    public static let asl: DDLoggerName // DDASLLogger
    public static let tty: DDLoggerName // DDTTYLogger
    public static let os: DDLoggerName // DDOSLogger
    public static let file: DDLoggerName // DDOSLogger

    public var isFileLogger: Bool { get }
}
```

Reply:
This is achievable by defining a C-function like that:
```
FOUNDATION_EXTERN inline BOOL DDLoggerNameIsFileLogger(DDLoggerName name) CF_SWIFT_NAME(getter:DDLoggerName.isFileLogger(self:));
```